Gems/Atom/Feature/Common/Code/Source/ImGui/ImGuiSystemComponent.cpp

```diff
@@ -141,15 +141,25 @@ namespace AZ
                 {
                     m_defaultImguiPassStack.erase(m_defaultImguiPassStack.begin() + i);
 
-                    // If the pass being removed as default is active, assume the current active should be changed
-                    // to whatever's on the top of the default stack as long as it has at least one entry.
+                    // If the pass being removed as default is at the top of the active stack, replace it
+                    // with whatever's now on the top of the default pass stack.
                     if (GetActiveContext() == imguiPass->GetContext())
                     {
                         PushActiveContextFromDefaultPass();
                     }
                     break;
                 }
             }
+
+            // The ImGuiPass will delete its context so we need to remove it from the active list
+            for (size_t i = 0; i < m_activeContextStack.size(); ++i)
+            {
+                if (imguiPass->GetContext() == m_activeContextStack.at(i))
+                {
+                    m_activeContextStack.erase(m_activeContextStack.begin() + i);
+                    break;
+                }
+            }
```

Comment:
Could reduce this with 
AZStd::erase_if(m_activeContextStack, [imguiPass](const auto& context) { return context == imguiPass->GetContext(); });